configure.ac

```diff
@@ -133,10 +133,21 @@ AM_CONDITIONAL(ANDROID, test x$ARCH = xandroid)
 
 AC_PROG_AWK
 
-AC_ARG_WITH(cxx, AS_HELP_STRING([--with-cxx=COMMAND], [Explicitly specify the C++ compiler to use]),[WITHCXX="$withval"],[WITHCXX="$CXX"])
+AC_MSG_CHECKING([for compiler override by OVERRIDE_CXX environment variable])
+if test "x$OVERRIDE_CXX" != "x"; then
+	AC_MSG_RESULT([yes, with $OVERRIDE_CXX])
```

Comment:
It might be better to use CXX directly and change the ci-linux workflow to use it instead, as @rofl0r mentioned.

Reply:
I hane taken a look into the generated code of the compiler choice macros 
```
AC_PROG_CXX([$cxx_compilers])·
·
AC_LANG([C++])·
```
It looks to me as if `$CXX` is indeed taken when set, but then it acts as a very quick exit bypassing all internal checks.

I would agree with the direct use of `$CXX` if we were using simple `make`, but we are not, we are using the Autotools and set the compiler and its options in a rather complex way. To say it otherwise, I look at the `$CXX` or the `$CXXFLAGS` as the output of the Autotools compiler macros and the input to the Makefiles, not the input of the Autotools.

Then, I felt that if the `ci-linux` workflow was using `$OVERRIDE_CXX` and not `$CXX` directly, which would have been taken immediately and would have provided CLang support to Linux Exult from the beginning, it must have been a good reason ... Am I wrong ?

Reply:
Thinking back, I think that I forgot to add either `--with-cxx=$(which $OVERRIDE_CXX)` or `CXX=$(which $OVERRIDE_CXX)` as arguments to `./configure`, the later being the "recommended" way of specifying a compiler in autotools. On a personal project which uses a similar setup (but with cmake instead of autotools), I export CXX before calling cmake to "configure", which is the recommended way of doing that with cmake.

For what is worth, I think that the logic of AC_PROG_CXX skipping most internal checks when you specify a compiler by hand is that when you do this, you are basically assuming the responsibility of ensuring that the compiler can do the job.

Reply:
Then why not simplify into :

1. `configure.ac` : remove `--with-cxx` and the handling of `$OVERRIDE_CXX` and let `make` / the Autotools acquire `CXX` either from the environment variable `$CXX` or from the `CXX=` option to `./configure`,
2. `ci-linux` : replace OVERRIDE_CXX by CXX and either stop there or pass `CXX=${CXX}` to './configure` ?

Reply:
That works, yes.